### ISSUE TITLE
fatal error: ThreadPool.h: No such file or directory

### ISSUE THREAD

**Issue by Darker** [log4cplus/log4cplus#169]
I get the above error when building log4cplus on Debian 8x32bit.  The full context of the error is:

```
make[2]: Entering directory '/root/cpp/log4cplus-master'
/bin/bash ./libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H   -I./include -I./threadpool -I./include -I./catch/single_include -DCATCH_CONFIG_PREFIX_ALL=1 -DNDEBUG -DINSIDE_LOG4CPLUS -D_GNU_SOURCE=1 -D_REENTRANT  -std=c++0x -Wall -fdiagnostics-show-caret -ftrack-macro-expansion -fdiagnostics-color=auto -Wextra -pedantic -Wstrict-aliasing -Wstrict-overflow -Woverloaded-virtual -Wold-style-cast -Wundef -Wshadow -Wformat -Wnoexcept -Wsuggest-attribute=format -Wsuggest-attribute=noreturn -Wno-variadic-macros -O2 -fvisibility=hidden -pthread -MT src/liblog4cplus_la-global-init.lo -MD -MP -MF src/.deps/liblog4cplus_la-global-init.Tpo -c -o src/liblog4cplus_la-global-init.lo `test -f 'src/global-init.cxx' || echo './'`src/global-init.cxx
libtool: compile:  g++ -DHAVE_CONFIG_H -I./include -I./threadpool -I./include -I./catch/single_include -DCATCH_CONFIG_PREFIX_ALL=1 -DNDEBUG -DINSIDE_LOG4CPLUS -D_GNU_SOURCE=1 -D_REENTRANT -std=c++0x -Wall -fdiagnostics-show-caret -ftrack-macro-expansion -fdiagnostics-color=auto -Wextra -pedantic -Wstrict-aliasing -Wstrict-overflow -Woverloaded-virtual -Wold-style-cast -Wundef -Wshadow -Wformat -Wnoexcept -Wsuggest-attribute=format -Wsuggest-attribute=noreturn -Wno-variadic-macros -O2 -fvisibility=hidden -pthread -MT src/liblog4cplus_la-global-init.lo -MD -MP -MF src/.deps/liblog4cplus_la-global-init.Tpo -c src/global-init.cxx  -fPIC -DPIC -o src/.libs/liblog4cplus_la-global-init.o
src/global-init.cxx:35:24: fatal error: ThreadPool.h: No such file or directory
 #include "ThreadPool.h"
                        ^
compilation terminated.
Makefile:2552: recipe for target 'src/liblog4cplus_la-global-init.lo' failed
make[2]: *** [src/liblog4cplus_la-global-init.lo] Error 1
make[2]: Leaving directory '/root/cpp/log4cplus-master'
Makefile:3619: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/root/cpp/log4cplus-master'
Makefile:1432: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by wilx** [comment]
You need to clone with `--recursive` parameter or `git submodule update --init --recursive` after clone.

**Comment by Darker** [comment]
I didn't clone, I downloaded master branch as zip archive.

**Comment by wilx** [comment]
@Darker Well, do not use that. Either use [tarballs available from SourceForge](https://sourceforge.net/projects/log4cplus/files/log4cplus-stable/) (but there are not any tarballs for master branch) or clone with `--recursive` parameter. This appears to be a GitHub issue that I cannot do anything about.

**Comment by Darker** [comment]
Thank you for your help, I'll try to do it as you suggested.

**Comment by ScaryAardvark** [comment]
I too had this issue. It may be worth updating your documentation to point out that you need to do a recursive clone :)

**Comment by 79rpm** [comment]
Thank you @wilx for updating the documentation.

**Comment by wilx** [comment]
Fixed by f5401762a5ee9b52d9a939dbabd11c014be8288c.
